pkg/maintainer/spv/chain.go

```diff
@@ -27,7 +27,30 @@ type Chain interface {
 		fundingOutputIndex uint32,
 	) (*tbtc.DepositChainRequest, error)
 
+	// TODO: Most likely merge it with `GetDepositRequest` function.
+	// Deposits gets the on-chain deposit request for the given funding
+	// transaction hash and output index. If the deposit does not exist it
+	// returns a zero-filled structure without an error.
```

Comment:
Agreed, those functions are mostly identical right now.

Reply:
As joining `GetDepositRequest` and `Deposits` modifies some code unrelated to this PR I decided to handle it in a small separate PR: https://github.com/keep-network/keep-core/pull/3654.
Once the PR is merged to main, I will remove the `Deposits` function.